hbase-server/src/main/java/org/apache/hadoop/hbase/mob/DefaultMobStoreCompactor.java

```diff
@@ -326,31 +474,29 @@ protected boolean performCompaction(FileDetails fd, InternalScanner scanner, Cel
       progress.cancel();
       throw new InterruptedIOException(
           "Interrupted while control throughput of compacting " + compactionName);
+    } catch (IOException t) {
+      LOG.error("Mob compaction failed for region: " + store.getRegionInfo().getEncodedName());
+      throw t;
     } finally {
       // Clone last cell in the final because writer will append last cell when committing. If
       // don't clone here and once the scanner get closed, then the memory of last cell will be
       // released. (HBASE-22582)
       ((ShipperListener) writer).beforeShipped();
       throughputController.finish(compactionName);
       if (!finished && mobFileWriter != null) {
+        // Remove all MOB references because compaction failed
+        mobRefSet.get().clear();
+        // Abort writer
```

Comment:
log at debug that we're aborting because we got a failure.

Reply:
Fixed.